megaqc/model/models.py

```diff
@@ -25,26 +24,33 @@ class Report(db.Model, CRUDMixin):
 
     __tablename__ = 'report'
     report_id = Column(Integer, primary_key=True)
-    user_id = Column(Integer, ForeignKey('users.user_id'), index=True)
+    # If the user is deleted, we still want to retain the report
+    user_id = Column(Integer, ForeignKey('users.user_id', ondelete='SET NULL'), index=True)
```

Comment:
I feel user accounts should be deactivated as opposed to deleted for historical purposes. This is good as a precaution ensure the report is retained in the event of an accidental or "accidental" deletion though.

Reply:
That's probably reasonable. But yes, that's not really what this snippet is doing. It's just ensuring database integrity if the user *is* somehow deleted.